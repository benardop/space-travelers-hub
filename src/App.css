@import url('https://fonts.googleapis.com/css2?family=Poppins&display=swap');

:root {
  --color-black: #212529;
  --color-blue: #157af6;
  --color-greenishblue: #42a2b9;
  --color-white: #fff;
}

.App {
  margin: 8rem 0 5rem 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}

html {
  scroll-behavior: smooth;
}

