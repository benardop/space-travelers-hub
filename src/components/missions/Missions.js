import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissionsApi } from '../../redux/missions/MissionsSlice';
import Missioncard from './Missioncard';
import '../stylesheets/Missioncard.css';

const Missionlist = () => {
  const dispatch = useDispatch();
  const mission = useSelector((state) => state.missions);
  useEffect(() => {
    if (!mission.length) dispatch(fetchMissionsApi());
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <div>
      <div className="heading">
        <div className="left">Mission</div>
        <div className="center">Description</div>
        <div className="right">Status</div>
      </div>
      { mission.map((item) => (
        <Missioncard
          key={item.missionid}
          missionname={item.missionName}
          description={item.missionDesc}
          id={item.missionid}
          missionjoin={item.missionjoin}
        />
      )) }
    </div>
  );
};

export default Missionlist;
