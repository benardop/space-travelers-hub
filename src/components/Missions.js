import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getMissionFromAPI } from '../redux/missions/missions';
import MissionCard from './MissionCard';

const Missions = () => {
  const dispatch = useDispatch();

  const missions = useSelector((state) => state.missionReducer);

  const mission = missions.missions.map((mission) => (
    <MissionCard key={mission.mission_id} mission={mission} />
  ));

  useEffect(() => {
    if (!missions.missions[0]) {
      dispatch(getMissionFromAPI());
    }
  }, []);

  return (
    <div className="contents">
      <table className="table table-striped table-hover">
        <thead>
          <tr>
            <th style={{ width: '10%' }} scope="col">Mission</th>
            <th style={{ width: '70%' }} scope="col">Description</th>
            <th style={{ width: '10%' }} scope="col">Status</th>
            <th style={{ width: '10%' }} scope="col">Action</th>
          </tr>
        </thead>
        {mission}
      </table>
    </div>
  );
};

export default Missions;
