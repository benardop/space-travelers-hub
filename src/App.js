import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from './redux/configureStore';
import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';
import Navbar from './components/Navbar';
import Rockets from './components/Rockets';
import Missions from './components/Missions';
import MyProfile from './components/MyProfile';

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Navbar />
        <Routes>
          <Route path="/" element={<Rockets />} />
          <Route path="/missions" element={<Missions />} />
          <Route path="/myprofile" element={<MyProfile />} />
        </Routes>
      </Router>
    </Provider>
  );
}

export default App;
