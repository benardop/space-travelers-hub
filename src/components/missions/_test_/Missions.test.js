import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Missions from '../Missions';
import store from '../../../redux/configureStore';
import App from '../../../App';

// missions snapshots
const MockMissions = () => (
  <Provider store={store}>
    <Missions />
  </Provider>
);

it('matches snapshot', () => {
  const tree = renderer.create(
    <MockMissions />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// test missions link from the app section
const MockApp = () => (
  <Provider store={store}>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>
);

describe('when you click the missions link', () => {
  it('should render missions page content', () => {
    render(
      (<MockApp />),
    );
    userEvent.click(screen.getByRole('link', { name: /missions/i }));
    expect(screen.getByText(/description/i)).toBeInTheDocument();
  });
});
