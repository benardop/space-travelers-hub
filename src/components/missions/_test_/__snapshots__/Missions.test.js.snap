// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="heading"
  >
    <div
      className="left"
    >
      Mission
    </div>
    <div
      className="center"
    >
      Description
    </div>
    <div
      className="right"
    >
      Status
    </div>
  </div>
</div>
`;
